### PR TITLE
[SPARK-41990][SQL] Use `FieldReference.column` instead of `apply` in V1 to V2 filter conversion

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?

Use `FieldReference.column` instead of `FieldReference.apply` in V1 to V2 filter conversion

### Why are the changes needed?

Previously, filtering by composite field name doesn't work

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @cloud-fan @sunchao @LuciferYang @panbingkun

**Review by dongjoon-hyun — COMMENTED** [review]
Could you fix two failures?
Screenshot 2023-01-13 at 9 28 23 PM

File: sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala [hunk 1]
```diff
@@ -91,7 +91,7 @@ case class EqualTo(attribute: String, value: Any) extends Filter {
   override def toV2: Predicate = {
     val literal = Literal(value)
     new Predicate("=",
-      Array(FieldReference(attribute), LiteralValue(literal.value, literal.dataType)))
+      Array(FieldReference.column(attribute), LiteralValue(literal.value, literal.dataType)))
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala:94]: The column name in v1 filter is generated by `PushableColumn`, and we need to match it.

When nested predicate pushdown is enabled, we already quote the column name if the name is not a valid SQL identifier. So `FieldReference.apply` is correct here. However, when it's disabled, the column name is the parsed name and `FieldReference.column` should be used.

I think what we can do is
```
try {
  FieldReference(attribute)
} catch {
  case _: ParseException => FieldReference.column(attribute)
}
```

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala:94]: We can add a `protected def toV2Column` in the base class of v1 filter

Comment (huaxingao) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala]: Thanks for the suggestion! Fixed.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 3.4.0. Thank you, @huaxingao and @cloud-fan .

**Comment by huaxingao** [comment]
Thanks @dongjoon-hyun @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala [hunk 2]
```diff
@@ -74,6 +75,15 @@ sealed abstract class Filter {
    * Converts V1 filter to V2 filter
    */
   private[sql] def toV2: Predicate
+
+  protected def toV2Column(attribute: String): NamedReference = {
+    try {
+      FieldReference(attribute)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala:81]: We should add some comments to explain it.

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala [hunk 3]
```diff
@@ -1963,4 +1977,9 @@ class JDBCSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-41990: Filter with composite name") {
+    val df = sql("SELECT * FROM composite_name WHERE `last name` = 'smith'")
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala:1982]: is it because we always disable nested predicate pushdown for JDBC v1?

Comment (huaxingao) [reply sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala:1982]: Right. Support for nested predicate pushdown is calculated by this [method](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala#L108), which returns false for JDBC V1.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala (hunk 2 above)
Comment (huaxingao) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala:81]: I will have a follow up to add some comments.

**Pull request closed without merging**